the-collab-lab/tcl-21-smart-shopping-list
# As vs item sort

## Description
This branch sorts grocery lists by how soon an item should be purchased. When two items have the same next purchase date, they are ordered alphabetically. Items are also color coded based on their state and contain description aria-labels for screen readers.

## Related Issue

Closes #13

## Acceptance Criteria

- Items in the list are shown as visually distinct (e.g., with a different background color on the list item) according to how soon the item is expected to be bought again: Soon, Kind of soon, Not soon, Inactive
- Items should be sorted by the estimated number of days until next purchase
- Items with the same number of estimated days until next purchase should be sorted alphabetically
- Items in the different states should be described distinctly when read by a screen reader

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates
### After
<img src="https://media.giphy.com/media/TL3E4mUumYqqX8H9Db/giphy.gif" />

## Testing Steps / QA Criteria

- From your terminal window pull down the related branch using the command `git pull origin as-vs-item-sort` and then migrate to that branch using `git checkout as-vs-item-sort`

- Run `npm install` && `npm start` to launch the server

- Navigate to the list view to your items sorted by how soon they will need to be purchased. Check that items that need to be bought soon are red, items that need to be bought kind of soon are yellow, items that do not need to be bought soon are green, and inactive items are grey.
